src/order/category/Lattice.lean

```diff
@@ -0,0 +1,66 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.category.PartialOrder
+import order.hom.lattice
+
+/-!
+# The category of lattices
+
+This defines `Lattice`, the category of lattices.
```

Comment:
It might be nice here to comment that you don't insist on a top and bottom, since oftentimes this is included in the definition of the category of lattices, eg here: https://ncatlab.org/nlab/show/lattice.